src/orion/core/worker/primary_algo.py

```diff
@@ -79,64 +119,133 @@ def suggest(self, num):
         if transformed_trials is None:
             return None
 
-        for trial in transformed_trials:
-            self._verify_trial(trial, space=self.transformed_space)
-            if trial not in self.transformed_space:
+        trials: list[Trial] = []
+        for transformed_trial in transformed_trials:
+            if transformed_trial not in self.transformed_space:
                 raise ValueError(
-                    f"Trial {trial.id} not contained in space:"
-                    f"\nParams: {trial.params}\n: Space{self.transformed_space}"
+                    f"Trial {transformed_trial.id} not contained in space:\n"
+                    f"Params: {transformed_trial.params}\n"
+                    f"Space: {self.transformed_space}"
                 )
-
-        trials = []
-        for transformed_trial in transformed_trials:
-            trial = self.transformed_space.reverse(transformed_trial)
-            self._verify_trial(trial, space=self.space)
-            trials.append(trial)
-
+            original = self.transformed_space.reverse(transformed_trial)
+            if original in self.registry:
+                # We already have a trial that is equivalent to this one.
+                # Copy over the status and results and observe it.
+                transformed_trial = _copy_status_and_results(
+                    original_trial=original, transformed_trial=transformed_trial
+                )
+                self.algorithm.observe([transformed_trial])
+            else:
+                # We haven't seen this trial before. Register it.
+                trials.append(original)
+
+            # Register the trial in our registry.
+            self.registry.register(original)
```

Comment:
If the original was already in register, we may overwrite the status of the previous one, losing results and completed status for instance. Unless I am not reading the code correctly and there is no issue, these cases should be tested explicitly to avoing reintroducing these issues.

Reply:
Good point. I'll only register the trial in the case where it wasn't already in the registry, just to be safe.

Reply:
Fixed in eebe53ca24d35db5b9cb917c3a319e4755d41266

Reply:
If we do this then we never get the trial result registered in the register and when we suggest duplicates of completed trials we will never observe these results because we ignored them when we observed the original trial (we did not register the trial with results since it was already in the registry). We should have tests for this situation.

Reply:
Sorry I got confused with `observe()`. Yes, what you propose is good here in `suggest()`, only register if the original was not already know.

Reply:
Can we mark this as resolved?